source/jormungandr/jormungandr/street_network/with_parking.py

```diff
@@ -0,0 +1,107 @@
+#
+# This file is part of Navitia,
+#     the software to build cool stuff with public transport.
+#
+# Hope you'll enjoy and contribute to this project,
+#     powered by Canal TP (www.canaltp.fr).
+# Help us simplify mobility and open public transport:
+#     a non ending quest to the responsive locomotion way of traveling!
+#
+# LICENCE: This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+# Stay tuned using
+# twitter @navitia
+# IRC #navitia on freenode
+# https://groups.google.com/d/forum/navitia
+# www.navitia.io
+
+import logging
+from jormungandr.street_network.street_network import AbstractStreetNetworkService
+from jormungandr import street_network
+
+from jormungandr.street_network.asgard import Asgard
+from jormungandr import utils
+
+
+class WithParking(AbstractStreetNetworkService):
+    def __init__(self, instance, service_url, modes=[], id='with_parking', timeout=10, api_key=None, **kwargs):
+        self.instance = instance
+        self.modes = modes
+        self.sn_system_id = id
+        self.parking_module = utils.create_object(kwargs.get('parking', None))
+        config = kwargs.get('street_network', None)
+        if 'service_url' not in config['args']:
+            config['args'].update({'service_url': None})
+        if 'instance' not in config['args']:
+            config['args'].update({'instance': instance})
+
+        self.street_network = utils.create_object(config)
+
+    def status(self):
+        return {'id': unicode(self.sn_system_id), 'class': self.__class__.__name__, 'modes': self.modes}
+
+    def _direct_path(
+        self, mode, pt_object_origin, pt_object_destination, fallback_extremity, request, direct_path_type
+    ):
+        response = self.street_network._direct_path(
+            mode, pt_object_origin, pt_object_destination, fallback_extremity, request, direct_path_type
+        )
+        if response and len(response.journeys):
+            self.add_parking_time_in_direct_path(response)
+
+        return response
+
+    def add_parking_time_in_direct_path(self, response):
+        logger = logging.getLogger(__name__)
+        logger.debug("Adding parking time for direct path " + str(self.parking_module.park_duration))
+        for journey in response.journeys:
+            journey.duration += self.parking_module.park_duration
```

Comment:
hmmm, don't you need to add parking section at the end of the journey?

Reply:
Shouldn't this being calculated by Asgard instead ?

Reply:
@xlqian Yes, I have to create a section in the response, not only adding the time. Will do now.

@nonifier The parking module and the calculator (Asgard, Kraken, Géovelo...) used are independent. The calculator doesn't know anything about the parking module.